### PR TITLE
Energy gun fixes

### LINKED ISSUE #5381: Ebow does not work
## Description
The traitor-spawned ebow does not work,in general. It appears unloaded,and refuses to generate new ammo.

### Steps to Reproduce

1.  Roll traitor
2.  Buy ebow
3.  Try to shoot ebow but fail miserably

### LINKED ISSUE #5477: Guns won't reload after shooting
1. shoot gun 
2. try to reload
3.observed lack of reloading

### LINKED ISSUE #5487: Gun bugs
A few errors I found in develop:

1- Clients cant fire energy guns unless they click on them while holding them in their hand
2- Clients dont sync the energy gun firemode with the server, ej: the clients see themselves shooting a disabler while the server and other clients shows a laser
3- Energy guns can get stuck and refuse to fire for any player
4- Sometimes a runtime pops out when a mob gets hit by a gun

### PR DESCRIPTION
**Pull request by Aranclanos**
### Purpose
Fixes #5381
Fixes #5477
Fixes #5487
Fixes energy crossbows not firing.
Fixes energy guns not changing firemode on all clients.
Fixes shotguns not being able to reload.
Sets the base energy weapon prefab to have a firing mode.
Fixes runtimes on changing firing mode on certain guns.
Fixes repeated firing modes on certain guns.
Fixes projectiles from not appearing on clients if others shot it.

### CODE REVIEW

**Comment by AutumnNova** [comment]
#5418 fixed all of this

File: UnityProject/Assets/Scripts/Items/Weapons/MagazineBehaviour.cs [hunk 1]
```diff
@@ -188,7 +188,7 @@ public virtual void ExpendAmmo(int amount = 1)
 	[Server]
 	public void ServerSetAmmoRemains(int remaining)
 	{
-		SyncServerAmmo(serverAmmoRemains, remaining);
```
Comment (AutumnNova) [inline UnityProject/Assets/Scripts/Items/Weapons/MagazineBehaviour.cs:193]: decently sure this goes against
https://unitystation.github.io/unitystation/development/SyncVar-Best-Practices-for-Easy-Networking/, same goes with all other removals of SyncServerAmmo()
```

[Server]
private void ServerChangeName(string newName)
{
    //NO! BAD! DON'T DO THIS!
    this.itemName = newName;

    //YES! GOOD! DO THIS INSTEAD!
    SyncItemName(newName);
}

```
this part to be exact

Comment (Aranclanos) [reply UnityProject/Assets/Scripts/Items/Weapons/MagazineBehaviour.cs:193]: thanks ill swap them

**Comment by Aranclanos** [comment]
Conflicts resolved

**Comment by Bod9001** [comment]
Sorry more merge conflicts

**Comment by Aranclanos** [comment]
solved

**Pull request merged**